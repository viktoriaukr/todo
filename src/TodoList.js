import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
  const [todos, setTodos] = useState([]);
  const addTodo = (newTodo) => {
    setTodos((todos) => [...todos, newTodo]);
  };
  const removeTodo = (id) =>
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  const newTodos = todos.map((todo) => (
    <Todo key={todo.id} id={todo.id} removeTodo={removeTodo} task={todo.task} />
  ));
  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul>{newTodos}</ul>
    </div>
  );
};

export default TodoList;
