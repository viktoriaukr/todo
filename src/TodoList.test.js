import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todo, task = "test todo") {
  const taskInput = todo.getByLabelText("Todo");
  fireEvent.change(taskInput, { target: { value: task } });
  const btn = todo.getByText("Add Todo");
  fireEvent.click(btn);
}

test("render", () => {
  render(<TodoList />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test("adding a new todo", () => {
  const todo = render(<TodoList />);
  expect(todo.queryByText("X")).not.toBeInTheDocument();
  addTodo(todo);
  expect(todo.queryByText("X")).toBeInTheDocument();
  expect(todo.getByText("test todo")).toBeInTheDocument();
});

test("removing todo", () => {
  const todo = render(<TodoList />);
  addTodo(todo);
  const btn = todo.getByText("X");
  fireEvent.click(btn);
  expect(btn).not.toBeInTheDocument();
});
