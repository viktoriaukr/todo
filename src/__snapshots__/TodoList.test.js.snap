// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Todo
      </label>
      <input
        id="task"
        name="task"
        placeholder="New Todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
